(blog) Status Update 2021-05-27 (#254)
puppetlabs/iac: _posts/2021-05-28-status-update.md

## New Module / Gem Releases

The following modules were released this week:

- [`puppetlabs-docker`][puppetlabs-docker] (`4.0.1`)
- [`puppetlabs-apache`][puppetlabs-apache] (`6.2.0`)
- [`puppetlabs-postgresql`][puppetlabs-postgresql] (`7.2.0`)
- [`puppetlabs-haproxy`][puppetlabs-haproxy] (`6.0.1`)
- [`puppetlabs-tomcat`][puppetlabs-tomcat] (`5.1.0`)
- [`puppetlabs-vsphere`][puppetlabs-vsphere] (`2.0.0`)

  [puppetlabs-docker]: https://github.com/puppetlabs/puppetlabs-docker
  [puppetlabs-apache]: https://github.com/puppetlabs/puppetlabs-apache
  [puppetlabs-postgresql]: https://github.com/puppetlabs/puppetlabs-postgresql
  [puppetlabs-haproxy]: https://github.com/puppetlabs/puppetlabs-haproxy
  [puppetlabs-tomcat]: https://github.com/puppetlabs/puppetlabs-tomcat
  [puppetlabs-vsphere]: https://github.com/puppetlabs/puppetlabs-vsphere
  [puppetlabs-mysql-pr-1387]: https://github.com/puppetlabs/puppetlabs-mysql/pull/1387
  [rsynnest]: https://github.com/rsynnest
  [iac-pr-252]: https://github.com/puppetlabs/iac/pull/252
  [dependabot]: https://github.com/apps/dependabot

### DSC And the PDK Go Prototype

> With the DevX team [hard at work on a new approach to PDK templating][pdkgo-templating] (and the release of their work imminent), this felt like a good week to see what it will be like to use the new templating for a real project.
> We put up a [prototype PR][puppet-dsc-pdkgo-template] for moving from switching to the new templating (if available) in our [`Puppet.Dsc`][puppet-dsc-intro-blog] module.

Reviewer: ```suggestion
The DevX team is [hard at work on a new approach to PDK templating][pdkgo-templating] and their LA release is imminent), so this week felt like a good time to see what it will be like to use the new templating in a real project.

We put up a [prototype PR][puppet-dsc-pdkgo-template] for switching to the new templating (when it becomes available) in our [`Puppet.Dsc`][puppet-dsc-intro-blog] module.
```

But when we make updates to the core builder or pwshlib library, we're actually rebuilding older module versions, which means even if we are extremely conservative with our estimations and pretend the PowerShell modules only average 5 releases each, we're going to shave a _frankly astounding_ **875 hours** from our build time!

The real time to rebuild and publish is, of course, shorter because we will be running this all in parallel; but saving more than a literal month's worth of build time on every release of `Puppet.Dsc` is an overwhelming win for us (and for the folks who maintain the Github Actions infrastructure)!

[pdkgo-templating]: https://github.com/puppetlabs/pdkgo
[puppet-dsc-autopublish]: https://github.com/puppetlabs/Puppet.Dsc/milestone/2
[puppet-dsc-forge]: https://forge.puppet.com/modules/dsc
[puppet-dsc-gallery]: https://www.powershellgallery.com/packages/Puppet.Dsc/0.5.0
[puppet-dsc-intro-blog]: https://puppetlabs.github.io/iac/news/roadmap/2020/09/21/dsc-release.html
[puppet-dsc-pdkgo-template]: https://github.com/puppetlabs/Puppet.Dsc/pull/154
[puppet-dsc-stable]: https://github.com/puppetlabs/Puppet.Dsc/milestone/1

> ### PDK Content Templates
> 
> You may have seen the PDK user experience survey that went around some weeks ago.
> One of the hypothesis that we started with was a prediction that the monolithic [pdk-templates][pdk-templates] repository was too complex and inflexible for end users.
> The [results][pdk-results] of the survey bore this out, and so we started work on a new approach to content templating.
> Instead of a single repository which you'd need to fork and maintain to keep up to date, we built out a modular system that would allow you pick and choose the components you'd like to include.
> I'm happy to report that we are making great progress and will have a limited availability release out very soon for you to experiment with so watch this space for updates!
> 
> [pdk-templates]: https://github.com/puppetlabs/pdk-templates
> [pdk-results]: https://docs.google.com/document/d/19LzpixC694nOklW6-IqwcGBzlwR6v56M-sMsdSS14Ss/edit#

Reviewer: ```suggestion
### Puppet Content Templates LA Release

You may have seen the PDK user experience survey that went around some weeks ago. We wanted to gain your feedback to validate some assumptions we had about creating content for Puppet products.  One of the hypothesis that we started with was a prediction that the monolithic [pdk-templates][pdk-templates] repository was too complex and inflexible for end users.

The [results][pdk-results] of the survey bore this out, and so we started work on a new approach to content templating.

Instead of a single repository which you'd need to fork and maintain to keep up to date, we built out a modular system that would allow you pick and choose the components you'd like to include.

We're happy to report that we are making great progress and will have a limited availability release out very soon for you to experiment with, so watch this space for updates!

[pdk-templates]: https://github.com/puppetlabs/pdk-templates
[pdk-results]: https://docs.google.com/document/d/19LzpixC694nOklW6-IqwcGBzlwR6v56M-sMsdSS14Ss/edit#
```

Reviewer: It makes sense for this section to go before the `Puppet.Dsc` section as it explains a bit what PCT are before Lombardi starts using them
